src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Item>(
+  items: Array<Item>,
+  key: keyof Item,
```

Comment:
```suggestion
export function groupByKey<T>(
  items: Array<T>,
  key: keyof T,
```
we usually just name it as `T`